File: C++/Intersection-of-Two-Linked-Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
	int getLength(ListNode *head) {
		int res = 0;
		while (head != NULL) {
			head = head->next;
			res ++;
		}
		return res;
	}
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        int lenA = getLength(headA), lenB = getLength(headB);
        int temp;
        if (lenA < lenB) {
        	temp = lenB - lenA;
        	while (temp --) {
        		headB = headB->next;
        	}
        } else {
        	temp = lenA - lenB;
        	while (temp --) {
        		headA = headA->next;
        	}
        }
        while (headA != NULL && headB != NULL) {
        	if (headA == headB) return headA;
        	headA = headA->next;
        	headB = headB->next;
        }
        return NULL;
    }
};

File: Python/Intersection-of-Two-Linked-Lists.py
# Definition for singly-linked list.
class ListNode:
     def __init__(self, x):
         self.val = x
         self.next = None

class Solution:
    # @param two ListNodes
    # @return the intersected ListNode
    def getIntersectionNode(self, headA, headB):
        lenA = 0
        lenB = 0
        nodeA = headA
        nodeB = headB
        while nodeA != None:
            lenA += 1
            nodeA = nodeA.next
        while nodeB != None:
            lenB += 1
            nodeB = nodeB.next
        nodeA = headA
        nodeB = headB
        if lenA > lenB:
            for i in range(lenA - lenB):
                nodeA = nodeA.next
        else:
            for i in range(lenB - lenA):
                nodeB = nodeB.next
        while nodeA != None and nodeB != None:
            if nodeA == nodeB:
                return nodeA
            else:
                nodeA = nodeA.next
                nodeB = nodeB.next
        return None